process-engine/process_engine_core
# 🐛 Fixes for EventHandling

## Changes

1. Fix null-value handling for BoundaryEvent callbacks
2. Fix timing of `ProcessInstanceFinished` notification
    - Publishing this notification through the EndEventHandler could have the effect of the notification occurring, before the ProcessInstance was actually finished. 
    - The notification is now published through the `ProcessInstanceStateHandlingFacade`, after the ProcessInstance's state was changed to `finished`
3. Properly rename some UseCases on the `ProcessInstanceStateHandlingFacade` and update corresponding tests
4. Add EndEvent results to ProcessTokenFacade

## Issues

PR: #314

## How to test the changes

- Start a ProcessInstance that makes use of `MessageBoundaryEvents` and/or `SignalBoundaryEvents`
- Trigger these events, using a payload that is `null` or `undefined`
- See that it works
